fix unit test case issues and other minor bugs

anithapriyanatarajan opened the pull request:
# Changes

<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

`config.go` - enhanced logging and validation
`controller.go`,`ttl_handler.go`,`reconciler.go` - fixed minor bugs and improved logging
`*_test.go` - fixed all failing and flaky unit test cases

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Run `make test lint` before submitting a PR
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

# Release Notes
```release-note
NONE
```

<!--
Describe any user facing changes here, or delete this block.

Examples of user facing changes:
- API changes
- Bug fixes
- Any changes in behavior
- Changes requiring upgrade notices or deprecation warnings

For pull requests with a release note:

```release-note
Your release note here
```

For pull requests that require additional action from users switching to the new release, include the string "action required" (case insensitive) in the release note:

```release-note
action required: your release note here
```

For pull requests that don't need to be mentioned at release time, use the `/release-note-none` Prow command to add the `release-note-none` label to the PR. You can also write the string "NONE" as a release note in your PR description:

```release-note
NONE
```
-->

pramodbindal commented on .github/workflows/kind-e2e.yaml:
```diff
@@ -38,7 +35,6 @@ jobs:
         # This should follow the list of versions from https://github.com/tektoncd/pipeline/blob/main/releases.md#release
-        - v0.62.2  # LTS
         - v0.65.2  # LTS
         - v0.70.0
```
Now 1.0.0 is release on pipelines to lets use that

pramodbindal commented:
/retest

pramodbindal commented:
@anithapriyanatarajan 
Can you make all workflows successful please

pramodbindal commented:
/retest

anithapriyanatarajan commented:
@pramodbindal - Thanks for your comments. I have addressed the concerns in https://github.com/openshift-pipelines/tektoncd-pruner/pull/202 and the uint test related fixes are submitted now separately now in https://github.com/openshift-pipelines/tektoncd-pruner/pull/204. Hence converting the PR To draft and marking as closed.
